File: 21d_python/3_año_bisiesto.py
def is_leap_year(year):
    # tu código aquí 👈
    if year >= 0 and type(year) == int:
        if year % 4 == 0:
            if year % 100 == 0:
                if year % 400 == 0:
                    return True
                else:
                    return False
            return True
        else:
            return False
    else:
        return False

print(is_leap_year(2000))
print(is_leap_year(-2024))
print(is_leap_year(1984.25))

"""
Otra forma de resolverlo:

def is_leap_year(year):
    if year % 1 != 0 or year <= 0:
      return False
    
    elif year % 4 == 0:
      if year % 100 == 0 and year % 400 == 0:
         return True

      elif year % 100 == 0: # else: return False
         return False
    
    else:
        return False
       
response = is_leap_year(2000)
print(response)
"""

File: 30d_JS/3_año_bisiesto.js
function isLeapYear(year){
    if (year >= 0 && year % 1 === 0){
        if (year % 4 === 0){
            if (year % 100 === 0){
                if (year % 400 === 0){
                    return(true);
                }
                else{
                    return(false);
                }
            }
            else{
                return(true);
            }
        }
        else{
            return(false);
        }
    }
    else{
        return(false);
    }
}

console.log(isLeapYear(2000));
console.log(isLeapYear(-2024));
console.log(isLeapYear(1984.25));